### PR TITLE
Implement API Guide For DSDK

### PR DESCRIPTION
**Pull request by 0Grit**
Initial Structure for DSDK Developer Guide

### CODE REVIEW

**Comment by 0Grit** [comment]
Change "Zephyr SDK" sectoin to "Zephyr"

**Comment by beriberikix** [comment]
Lots of great progress! Any idea why the preview isn't rendering?

**Comment by 0Grit** [comment]
> Lots of great progress! Any idea why the preview isn't rendering?

Not sure. It does render in gitpod though.

**Comment by 0Grit** [comment]
@vancefarren  Taking notes per our recent dicussion of the DFU guide.

* We will map out the sections of the guide we talked about
       * Guide-through of a golioth dfu example
       * Background concept of DFU
       * Golioth gut's mapped to the background concepts
       

* We will mark the 2nd two sections as Under construction this sprint
* We will complete the Guide-through this sprint.

**Comment by beriberikix** [comment]
Can we split the remaining content into another issue/PR so we don't lost them? Once reviewed we should push this!

File: docs/partials/coming-soon-centered.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# <center>🚧 Work in Progress... 🚧</center>
```
Comment (beriberikix) [inline docs/partials/coming-soon-centered.md:1]: This looks MASSIVE when rendered as an H1. I'd rather see this as normal text or maybe an [Admonition](https://docusaurus.io/docs/markdown-features/admonitions)

File: docs/firmware/device-sdk/firmware-upgrade/guide-dfu.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
+---
+title: 'Guide - Firmware Update'
```
Comment (beriberikix) [inline docs/firmware/device-sdk/firmware-upgrade/guide-dfu.md:2]: I'm not a fan of this title structure, ex. "Guide - " & "Concept - ". It adds verbosity.

Let's follow the convention from the Cloud section. Either "Firmware Update Overview" or simply "Overview." The nav hierarchy already provides the context.

I also realize the cloud section is inconsistent and we should clean that up separately.

File: docs/firmware/device-sdk/authentication/concepts-auth.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+title: 'Concepts - Device Auth'
```
Comment (beriberikix) [inline docs/firmware/device-sdk/authentication/concepts-auth.md:2]: These are fine to leave for now but would like them to drop the "Concepts - " and be more descriptive, like "Sending logs."

File: docs/firmware/device-sdk/authentication/guide-auth.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+title: 'Guide - Device Auth'
```
Comment (beriberikix) [inline docs/firmware/device-sdk/authentication/guide-auth.md:2]: Simply "Overview"; see other comment.

File: docs/firmware/device-sdk/firmware-upgrade/concepts-dfu.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+title: 'Concepts - Firmware Upgrade'
```
Comment (beriberikix) [inline docs/firmware/device-sdk/firmware-upgrade/concepts-dfu.md:2]: These are fine to leave for now but would like them to drop the "Concepts - " and be more descriptive, like "Sending logs."

File: docs/firmware/device-sdk/light-db-stream/concepts-lightdb-stream.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+title: Concepts - LightDB Stream
```
Comment (beriberikix) [inline docs/firmware/device-sdk/light-db-stream/concepts-lightdb-stream.md:2]: These are fine to leave for now but would like them to drop the "Concepts - " and be more descriptive, like "Sending logs."

File: docs/firmware/device-sdk/light-db-stream/guide-lightdb-stream.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
+title: Guide - LightDB Stream Client
```
Comment (beriberikix) [inline docs/firmware/device-sdk/light-db-stream/guide-lightdb-stream.md:2]: "Overview"; see other comment.

File: docs/firmware/device-sdk/light-db/concepts-lightdb.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+---
+title: Concepts - LightDB
```
Comment (beriberikix) [inline docs/firmware/device-sdk/light-db/concepts-lightdb.md:2]: These are fine to leave for now but would like them to drop the "Concepts - " and be more descriptive, like "Sending logs."

File: docs/firmware/device-sdk/light-db/guide-lightdb.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+---
+title: Guide - LightDB Client
```
Comment (beriberikix) [inline docs/firmware/device-sdk/light-db/guide-lightdb.md:2]: Overview; see other comment.

File: docs/firmware/device-sdk/logging/concepts-logging.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+---
+title: Concepts - Logging
```
Comment (beriberikix) [inline docs/firmware/device-sdk/logging/concepts-logging.md:2]: These are fine to leave for now but would like them to drop the "Concepts - " and be more descriptive, like "Sending logs."

File: docs/firmware/device-sdk/logging/guide-logging.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+---
+title: Guide - Logging Client
```
Comment (beriberikix) [inline docs/firmware/device-sdk/logging/guide-logging.md:2]: Overview; see other comment

File: docs/firmware/firmware-home.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+sidebar_position: 1
+---
+
+import ComingSoon from '/docs/partials/coming-soon-centered.md'
+
+<ComingSoon/>
+
+## Introduction
+
+Golioth is a cloud services platform for IoT devices. The accepted model for understanding
```
Comment (beriberikix) [inline docs/firmware/firmware-home.md:13]: Incomplete sentence?

File: docs/firmware/firmware-home.md [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+
+<ComingSoon/>
+
+## Introduction
+
+Golioth is a cloud services platform for IoT devices. The accepted model for understanding 
+
+## Firmware
+
+The Golioth firmware version is dependent upon the Device SDK you are using. Our first SDK is built on top of Zephyr
```
Comment (beriberikix) [inline docs/firmware/firmware-home.md:17]: Incomplete sentence?

File: docs/partials/coming-soon-centered.md (hunk 1 above)
Comment (0Grit) [reply docs/partials/coming-soon-centered.md:1]: Will try admonition

File: docs/firmware/device-sdk/firmware-upgrade/concepts-dfu.md (hunk 5 above)
Comment (0Grit) [reply docs/firmware/device-sdk/firmware-upgrade/concepts-dfu.md:2]: More "sections" per service, but smaller pages?

**Comment by 0Grit** [comment]
image

Per above image. Concepts can be covered in the overview. The 'guide" would then consist of pages with "action oriented"  titles.

Can not argue against what I myself consider a gold standard for documentation

File: docs/partials/coming-soon-centered.md (hunk 1 above)
Comment (0Grit) [reply docs/partials/coming-soon-centered.md:1]: image

**Pull request merged**